src/node.cc

```diff
@@ -303,6 +304,29 @@ MaybeLocal<Value> StartExecution(Environment* env, StartExecutionCallback cb) {
     first_argv = env->argv()[1];
   }
 
+#ifndef DISABLE_SINGLE_EXECUTABLE_APPLICATION
+  if (IsSingleExecutable()) {
+    size_t single_executable_application_size = 0;
+    const char* single_executable_application_code =
+        FindSingleExecutableCode(&single_executable_application_size);
+    if (single_executable_application_code != nullptr) {
+      Local<Value> buffer =
+          Buffer::New(
```

Comment:
Why are we using a Buffer here? I think this with the `toString()` in the main script would result in a copy, which is probably unnecessary.  It would be nicer to use an external string for this.

Reply:
Also I think `Buffer::New()` in this variant takes ownership of the data, that probably is not supposed to happen?